src/intersections/lines.jl

```diff
@@ -43,14 +28,43 @@ end
 function intersectpoint(l1::Line, l2::Line)
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
+  λ₁, _ = intersectparameters(a, b, c, d)
+  a + λ₁ * (b - a)
+ end
+
+#= 
+Compute the parameters of the lines defined by the points a -> b and c -> d 
+so that the resulting points have minimal distance.
+
+The ranks help to identify the different types of intersection:
 
-  v1  = a - b
-  v2  = c - d
-  v12 = a - c
+Intersection: r == rₐ == 2
+Collinear: r == rₐ == 1
+No intersection: r != rₐ
+  No intersection and parallel:  r == 1, rₐ == 2
+  No intersection, skew lines: r == 2, rₐ == 3
+=#
+function intersectparameters(a::Point{N,T}, b::Point{N,T}, 
+                             c::Point{N,T}, d::Point{N,T}) where {N,T}
+  # solves the equation (approximately):
+  # a + λ₁ ⋅ v⃗₁ = c + λ₂ ⋅ v⃗₂,
+  # with v⃗₁ = b - a, v⃗₂ = d - c
+  A = [(b - a) (c - d)]
+  y = c - a
+  QRₐ  = qr([A y])
+
+  # calculate the rank of the augmented matrix
+  rₐ = N == 2 ? sum(sum(abs, QRₐ.R; dims = 2) .> atol(T)) :
+                sum(abs.(diag(QRₐ.R)) .> atol(T))
```

Comment:
Can you please double check that the QR factorization object doesn't have this information stored already? I find it strange that we need to compute the rank with these summations.

Reply:
I've tried and benchmarked a lot of thinks:
- using pivoting in the QR decomposition (would make distinction between 2d and nd case obsolete)
  - slows down by a factor of 20 (no matter whether static or normal array)
- using rank directly
  - slows down significantly (involves svd decomposition)
- converting static array to normal array
  - makes things generally worse
- going directly for `A\y`
  - slows down significantly in `N > 2`

Regarding double checking:
rank for QR objects might be implemented in future:
see comment in [source code of qr.jl ](https://github.com/JuliaLang/julia/blob/master/stdlib/LinearAlgebra/src/qr.jl#L1058)

Reply:
Ok, now please refactor the summations as I had suggested before. These are all norms hidden as summations. Can you try expressing these expressions with `norm(x, p)`?